File: src/Asset/Vns.Erp.Core.Asset.CsAssetGUI/DanhMuc/frmDanhMucNguonVon_tscd_ChiTiet.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.IO;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using System.ComponentModel;

using Vns.Erp.Core;
using Vns.Erp.Core.Asset.Domain;
using Vns.Erp.Core.Asset.Service.Interface;

namespace Vns.Erp.Core.Asset.CsAssetGUI
{
    public partial class frmDanhMucNguonVon_tscd_ChiTiet : FrmBaseDmChiTiet
    {

        #region "Variables"
        private DmNguonTscd objNguonVon = new DmNguonTscd();
        //Private objNguonVonBO As IDmBtKetchuyenService
        //Public Property DmBtKetchuyenService() As IDmBtKetchuyenService
        //    Get
        //        Return objNguonVonBO
        //    End Get
        //    Set(ByVal value As IDmBtKetchuyenService)
        //        objNguonVonBO = value
        //    End Set
        //End Property


        IDmNguonTscdService _DmNguonTscdService;
        public IDmNguonTscdService DmNguonTscdService
        {
            get { return _DmNguonTscdService; }
            set { _DmNguonTscdService = value; }
        }

        private FormGlobals.DataInputState m_InputState;
        private string m_NV_tscd;
        #endregion
        private string p_Ma_Nguon_TSCD;

        #region "Funtions"

        #region "Overrides Function and Property"
        public frmDanhMucNguonVon_tscd_ChiTiet()
        {
            InitializeComponent();
            this.Load +=new EventHandler(frmDanhMucNguonVon_tscd_ChiTiet_Load);
        }
        public DmNguonTscd ObjectInfo
        {
            get
            {
                if ((ID != null))
                {
                    objNguonVon = DmNguonTscdService.Get(VnsConvert.CGuid(ID));
                }
                else
                {
                    objNguonVon = new DmNguonTscd();
                }
                objNguonVon.MaDmNguonTscd = txtMaTSCD.Text;
                objNguonVon.TenDmNguonTscd = txtTenTSCD.Text;
                objNguonVon.KyHieu = txtKYHIEU.Text;
                objNguonVon.MoTa = txtMOTA.Text;
                objNguonVon.DonviId = Generals.DON_VI.Id;
                objNguonVon.MaDonvi = Generals.DON_VI.MaDonvi;
                objNguonVon.CoSuDung = TNCommon.ParseDecimal(chkCO_SU_DUNG.Checked);
                objNguonVon.ChoPhepSua = TNCommon.ParseDecimal(chkCHO_PHEP_SUA.Checked);
                base.ResultObject = objNguonVon;
                return objNguonVon;
            }
            set
            {
                if ((value != null))
                {
                    objNguonVon = value;
                    p_Ma_Nguon_TSCD = objNguonVon.MaDmNguonTscd;
                    txtMaTSCD.Text = objNguonVon.MaDmNguonTscd;
                    txtTenTSCD.Text = objNguonVon.TenDmNguonTscd;
                    txtKYHIEU.Text = objNguonVon.KyHieu;
                    txtMOTA.Text = objNguonVon.MoTa;
                    chkCO_SU_DUNG.Checked = TNCommon.ParseBool(objNguonVon.CoSuDung);
                    chkCHO_PHEP_SUA.Checked = TNCommon.ParseBool(objNguonVon.ChoPhepSua);
                }
            }
        }

        public override void InitProperty()
        {
            Title = "Danh mục nguồn tài sản cố định";
            TableName = "DmNguonTscd";
            //RequireInputControls = New Control() {txtMaTSCD, txtKYHIEU, txtTenTSCD}
            ButtonSave = btnSave;
            ButtonClear = btnClear;
            ButtonCancel = btnCancel;
            //GroupControlObjectInfo = _GroupControlObjectInfo
            TNCommon.setIconControl(this);
        }

        public override void InitComponet()
        {
            //InitializeComponent() ' Not Use because of using in the design view
        }

        public override void BindLookUpEdit()
        {
            switch (Mode)
            {
                case FormGlobals.DataInputState.AddMode:
                    ClearObjectToControl();
                    break;
            }

        }

        public override void ValidateData(object sender, CancelEventArgs e)
        {
            //If txtMaTSCD.Text.Trim = "" Then
            //    Message_Warning("Bạn chưa nhập mã nguồn TSCD")
            //    txtMaTSCD.Focus()
            //    Exit Sub
            //End If

            //If txtKYHIEU.Text.Trim = "" Then
            //    Message_Information("Bạn chưa nhập ký hiệu nguồn TSCD ")
            //    txtKYHIEU.Focus()
            //    Exit Sub
            //End If

            //If txtTenTSCD.Text.Trim = "" Then
            //    Message_Information("Bạn chưa nhập tên nguồn TSCD ")
            //    txtTenTSCD.Focus()
            //    Exit Sub
            //End If
        }

        public void Save()
        {
            try
            {
                DmNguonTscd info = ObjectInfo;
                switch (Mode)
                {
                    case FormGlobals.DataInputState.AddMode:
                        ID = DmNguonTscdService.Save(info).Id;
                        break;
                    case FormGlobals.DataInputState.EditMode:
                        DmNguonTscdService.SaveOrUpdate(info);
                        break;
                }
                AfterSaveSuccess();
                this.DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }

        public override void BindData()
        {
            try
            {
                ObjectInfo = DmNguonTscdService.GetById(VnsConvert.CGuid(ID));
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }

        #endregion

        private void frmDanhMucNguonVon_tscd_ChiTiet_Load(object sender, EventArgs e)
        {
            InitProperty();
            CommonInit();
        }

        private void ClearObjectToControl()
        {
            objNguonVon = new DmNguonTscd();
            m_NV_tscd = Guid.NewGuid().ToString();
            p_Ma_Nguon_TSCD = objNguonVon.MaDmNguonTscd;
            txtMaTSCD.Text = objNguonVon.MaDmNguonTscd;
            txtTenTSCD.Text = objNguonVon.TenDmNguonTscd;
            txtKYHIEU.Text = objNguonVon.KyHieu;
            txtMOTA.Text = objNguonVon.MoTa;
            chkCO_SU_DUNG.Checked = false;
            chkCHO_PHEP_SUA.Checked = false;
        }

        private void SetObjectToControl()
        {
            objNguonVon = DmNguonTscdService.GetById(new Guid(m_NV_tscd));
            p_Ma_Nguon_TSCD = objNguonVon.MaDmNguonTscd;
            txtMaTSCD.Text = objNguonVon.MaDmNguonTscd;
            txtTenTSCD.Text = objNguonVon.TenDmNguonTscd;
            txtKYHIEU.Text = objNguonVon.KyHieu;
            txtMOTA.Text = objNguonVon.MoTa;
            chkCO_SU_DUNG.Checked = TNCommon.ParseBool(objNguonVon.CoSuDung);
            chkCHO_PHEP_SUA.Checked = TNCommon.ParseBool(objNguonVon.ChoPhepSua);
        }

        private bool Check_ma(string Ma_Nguon)
        {
            IList<DmNguonTscd> dt = DmNguonTscdService.GetByMaNguon(Ma_Nguon, Generals.DON_VI.Id);
            if (dt.Count != 0)
            {
                FormGlobals.Message_Warning("Mã nguồn TSCD đã tồn tại");
                txtMaTSCD.Focus();
                return false;
            }
            return true;
        }

        #endregion
        public override void Saving()
        {
            try
            {
                if (string.IsNullOrEmpty(txtMaTSCD.Text.Trim()))
                {
                    FormGlobals.Message_Warning("Bạn chưa nhập mã nguồn TSCD");
                    txtMaTSCD.Focus();
                    return;
                }

                if (string.IsNullOrEmpty(txtKYHIEU.Text.Trim()))
                {
                    FormGlobals.Message_Information("Bạn chưa nhập ký hiệu nguồn TSCD ");
                    txtKYHIEU.Focus();
                    return;
                }

                if (string.IsNullOrEmpty(txtTenTSCD.Text.Trim()))
                {
                    FormGlobals.Message_Information("Bạn chưa nhập tên nguồn TSCD ");
                    txtTenTSCD.Focus();
                    return;
                }
                Save();
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }
       
    }
}

File: src/Asset/Vns.Erp.Core.Asset.VbAssetGUI/DanhMuc/frmDanhMucNguonVon_tscd_ChiTiet.vb
Imports System
Imports System.Data
Imports System.IO
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports System.ComponentModel

Imports Vns.Erp.Core.Asset.Domain
Imports Vns.Erp.Core.Asset.Service.Interface

Public Class frmDanhMucNguonVon_tscd_ChiTiet
    Inherits FrmBaseDmChiTiet

#Region "Variables"
    Private objNguonVon As DmNguonTscd = New DmNguonTscd()
    'Private objNguonVonBO As IDmBtKetchuyenService
    'Public Property DmBtKetchuyenService() As IDmBtKetchuyenService
    '    Get
    '        Return objNguonVonBO
    '    End Get
    '    Set(ByVal value As IDmBtKetchuyenService)
    '        objNguonVonBO = value
    '    End Set
    'End Property

    Dim _DmNguonTscdService As IDmNguonTscdService

    Public Property DmNguonTscdService() As IDmNguonTscdService
        Get
            Return _DmNguonTscdService
        End Get
        Set(ByVal value As IDmNguonTscdService)
            _DmNguonTscdService = value
        End Set
    End Property

    Private m_InputState As FormGlobals.DataInputState
    Private m_NV_tscd As String
    Private p_Ma_Nguon_TSCD As String
#End Region

#Region "Funtions"

#Region "Overrides Function and Property"

    Public Property ObjectInfo() As DmNguonTscd
        Get
            If Not ID Is Nothing Then
                objNguonVon = DmNguonTscdService.Get(ID)
            Else
                objNguonVon = New DmNguonTscd()
            End If
            objNguonVon.MaDmNguonTscd = txtMaTSCD.Text
            objNguonVon.TenDmNguonTscd = txtTenTSCD.Text
            objNguonVon.KyHieu = txtKYHIEU.Text
            objNguonVon.MoTa = txtMOTA.Text
            objNguonVon.DonviId = Generals.DON_VI.Id
            objNguonVon.MaDonvi = Generals.DON_VI.MaDonvi
            objNguonVon.CoSuDung = TNCommon.ParseDecimal(chkCO_SU_DUNG.Checked)
            objNguonVon.ChoPhepSua = TNCommon.ParseDecimal(chkCHO_PHEP_SUA.Checked)
            MyBase.ResultObject = objNguonVon
            Return objNguonVon
        End Get
        Set(ByVal value As DmNguonTscd)
            If Not value Is Nothing Then
                objNguonVon = value
                p_Ma_Nguon_TSCD = objNguonVon.MaDmNguonTscd
                txtMaTSCD.Text = objNguonVon.MaDmNguonTscd
                txtTenTSCD.Text = objNguonVon.TenDmNguonTscd
                txtKYHIEU.Text = objNguonVon.KyHieu
                txtMOTA.Text = objNguonVon.MoTa
                chkCO_SU_DUNG.Checked = TNCommon.ParseBool(objNguonVon.CoSuDung)
                chkCHO_PHEP_SUA.Checked = TNCommon.ParseBool(objNguonVon.ChoPhepSua)
            End If
        End Set
    End Property

    Public Overrides Sub InitProperty()
        Title = "Danh mục nguồn tài sản cố định"
        TableName = "DmNguonTscd"
        'RequireInputControls = New Control() {txtMaTSCD, txtKYHIEU, txtTenTSCD}
        ButtonSave = btnSave
        ButtonClear = btnClear
        ButtonCancel = btnCancel
        'GroupControlObjectInfo = _GroupControlObjectInfo
        TNCommon.setIconControl(Me)
    End Sub

    Public Overrides Sub InitComponet()
        'InitializeComponent() ' Not Use because of using in the design view
    End Sub

    Public Overrides Sub BindLookUpEdit()
        Select Case Mode
            Case FormGlobals.DataInputState.AddMode
                ClearObjectToControl()
        End Select

    End Sub

    Public Overrides Sub ValidateData(ByVal sender As Object, ByVal e As CancelEventArgs)
        'If txtMaTSCD.Text.Trim = "" Then
        '    Message_Warning("Bạn chưa nhập mã nguồn TSCD")
        '    txtMaTSCD.Focus()
        '    Exit Sub
        'End If

        'If txtKYHIEU.Text.Trim = "" Then
        '    Message_Information("Bạn chưa nhập ký hiệu nguồn TSCD ")
        '    txtKYHIEU.Focus()
        '    Exit Sub
        'End If

        'If txtTenTSCD.Text.Trim = "" Then
        '    Message_Information("Bạn chưa nhập tên nguồn TSCD ")
        '    txtTenTSCD.Focus()
        '    Exit Sub
        'End If
    End Sub

    Public Sub Save()
        Try
            Dim info As DmNguonTscd = ObjectInfo
            Select Case Mode
                Case FormGlobals.DataInputState.AddMode
                    ID = DmNguonTscdService.Save(info).Id
                Case FormGlobals.DataInputState.EditMode
                    DmNguonTscdService.SaveOrUpdate(info)
            End Select
            AfterSaveSuccess()
            Me.DialogResult = System.Windows.Forms.DialogResult.OK
            Me.Close()
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub

    Public Overrides Sub BindData()
        Try
            ObjectInfo = DmNguonTscdService.GetById(ID)
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub

#End Region

    Private Sub frmDanhMucNguonVon_tscd_ChiTiet_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
        InitProperty()
        CommonInit()
    End Sub

    Private Sub ClearObjectToControl()
        objNguonVon = New DmNguonTscd()
        m_NV_tscd = Guid.NewGuid().ToString
        p_Ma_Nguon_TSCD = objNguonVon.MaDmNguonTscd
        txtMaTSCD.Text = objNguonVon.MaDmNguonTscd
        txtTenTSCD.Text = objNguonVon.TenDmNguonTscd
        txtKYHIEU.Text = objNguonVon.KyHieu
        txtMOTA.Text = objNguonVon.MoTa
        chkCO_SU_DUNG.Checked = False
        chkCHO_PHEP_SUA.Checked = False
    End Sub

    Private Sub SetObjectToControl()
        objNguonVon = DmNguonTscdService.GetById(New Guid(m_NV_tscd))
        p_Ma_Nguon_TSCD = objNguonVon.MaDmNguonTscd
        txtMaTSCD.Text = objNguonVon.MaDmNguonTscd
        txtTenTSCD.Text = objNguonVon.TenDmNguonTscd
        txtKYHIEU.Text = objNguonVon.KyHieu
        txtMOTA.Text = objNguonVon.MoTa
        chkCO_SU_DUNG.Checked = TNCommon.ParseBool(objNguonVon.CoSuDung)
        chkCHO_PHEP_SUA.Checked = TNCommon.ParseBool(objNguonVon.ChoPhepSua)
    End Sub

    Private Function Check_ma(ByVal Ma_Nguon As String) As Boolean
        Dim dt As IList(Of DmNguonTscd) = DmNguonTscdService.GetByMaNguon(Ma_Nguon, Generals.DON_VI.Id)
        If dt.Count <> 0 Then
            Message_Warning("Mã nguồn TSCD đã tồn tại")
            txtMaTSCD.Focus()
            Return False
        End If
        Return True
    End Function

#End Region

    Private Sub btnSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSave.Click
        Try
            If txtMaTSCD.Text.Trim = "" Then
                Message_Warning("Bạn chưa nhập mã nguồn TSCD")
                txtMaTSCD.Focus()
                Exit Sub
            End If

            If txtKYHIEU.Text.Trim = "" Then
                Message_Information("Bạn chưa nhập ký hiệu nguồn TSCD ")
                txtKYHIEU.Focus()
                Exit Sub
            End If

            If txtTenTSCD.Text.Trim = "" Then
                Message_Information("Bạn chưa nhập tên nguồn TSCD ")
                txtTenTSCD.Focus()
                Exit Sub
            End If
            Save()
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub
End Class
